XRTK-Core/Packages/com.xrtk.core/Runtime/Providers/Controllers/Simulation/Hands/SimulatedHandControllerDataProvider.cs

```diff
@@ -25,25 +26,54 @@ public class SimulatedHandControllerDataProvider : BaseSimulatedControllerDataPr
         public SimulatedHandControllerDataProvider(string name, uint priority, SimulatedHandControllerDataProviderProfile profile, IMixedRealityInputSystem parentService)
             : base(name, priority, profile, parentService)
         {
-            var poseDefinitions = profile.PoseDefinitions;
-            handPoseDefinitions = new List<SimulatedHandControllerPoseData>(poseDefinitions.Count);
+            var globalSettingsProfile = MixedRealityToolkit.Instance.ActiveProfile.InputSystemProfile;
```

Comment:
Lets just set the `profile` reference to global if `profile == null`. That way we can clean up all this duplicate code.

Reply:
My initial suggestion wouldn't work since `profile` and `globalProfile` are different types entirely'